### PR TITLE
Determine DC/OS version from leader's endpoint

### PR DESCRIPTION
**Pull request by gabrielhartmann**

### CODE REVIEW

**Comment by mbdcos** [comment]
@gabrielhartmann Is this to determine whether cluster is DCOS 1.9 or 1.10?

**Comment by gabrielhartmann** [comment]
@mbdcos: yes

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/DcosConstants.java [hunk 1]
```diff
@@ -8,11 +8,12 @@
...
 public class DcosConstants {
-    private static final String MASTER_MESOS = "master.mesos";
+    private static final String MESOS_MASTER = "master.mesos";
+    private static final String MESOS_LEADER = "leader.mesos";
 
     @SuppressWarnings("PMD.AvoidUsingHardCodedIP")
-    public static final String MESOS_MASTER_ZK_CONNECTION_STRING = MASTER_MESOS + ":2181";
-    public static final String MESOS_MASTER_URI = "http://" + MASTER_MESOS;
+    public static final String MESOS_MASTER_ZK_CONNECTION_STRING = MESOS_MASTER + ":2181";
+    public static final String MESOS_LEADER_URI = "http://" + MESOS_LEADER;
```
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/DcosConstants.java]: Isn't this HTTPS in strict? 🤔

Comment (gabrielhartmann) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/DcosConstants.java]: I don't know.  This has worked for a very long time.  Changing to `https` sounds like a different PR to me.

**Comment by gabrielhartmann** [comment]
retest this please

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/DcosConstants.java (hunk 1 above)
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/DcosConstants.java]: Hokay.

**Comment by gabrielhartmann** [comment]
retest this please

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/DcosConstants.java (hunk 1 above)
Comment (spahl) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/DcosConstants.java]: yes it will have to be https in strict and if possible in permissive too

Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/DcosConstants.java]: @spahl Does a redirect happen? We have had it this way for awhile and it hasn't had any issues.

**Pull request merged**